crates/interpreter/src/exec.rs

```diff
@@ -1423,6 +1406,158 @@ impl<'m> Thread<'m> {
         self.frames.push(Frame::new(inst_id, t.results.len(), ret, locals));
         Ok(ThreadResult::Continue(self))
     }
+
+    #[cfg(feature = "threads")]
+    fn atomic_notify(&mut self, _mem: &mut Memory<'m>, _m: MemArg) -> Result<(), Error> {
+        let count = self.pop_value();
+        let i = self.pop_value().unwrap_i32();
+
+        // Trap if memory access is OOB.
+        let _mem = match self.mem_slice(_mem, _m, i, 4) {
+            None => return Err(trap()),
+            Some(x) => x,
```

Comment:
Yes, we will certainly need some data for waiting threads. Maybe something like [parking_lot](https://docs.rs/parking_lot_core/latest/parking_lot_core/) would be handy. I have added a TODO for now.

Reply:
Ah yes, I'm mistaken. I somehow thought parking-lot was non-std and would allow you to pass in a synchronization primitive depending on the platform.

I will have to think about this a bit more and get back to you. Probably best to start simple though before generalizing further.